### PR TITLE
Add ROCm limitations to docs

### PR DESCRIPTION
**Pull request by kmaehashi**
@takagi could you fill these sections?

### CODE REVIEW

**Comment by takagi** [comment]
Sure!

**Comment by takagi** [comment]
(Let me clean up the commit log soon...)

**Comment by takagi** [comment]
@kmaehashi Would you take a look?

File: docs/source/install.rst [hunk 1]
```diff
@@ -425,3 +425,41 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
   This behavior is specific to ROCm builds; when building CuPy for NVIDIA CUDA, the build result is not affected by the host configuration.
+
+Limitations
+-----------
+
+The following CUDA-specific features are not available in ROCm:
+
+* CUDA Array Interface
+* cuTENSOR
+* Atomic addition in FP16 (``cupy.ndarray.scatter_add`` and `cupyx.scatter_add`)
```
Comment (kmaehashi) [inline docs/source/install.rst:436]:
```suggestion
* Atomic addition in FP16 (``cupy.ndarray.scatter_add`` and ``cupyx.scatter_add``)
```

File: docs/source/install.rst [hunk 2]
```diff
@@ -425,3 +425,41 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
 
   If you don't specify the ``HCC_AMDGPU_TARGET`` environment variable, CuPy will be built for the GPU architectures available on the build host.
   This behavior is specific to ROCm builds; when building CuPy for NVIDIA CUDA, the build result is not affected by the host configuration.
+
+Limitations
+-----------
+
+The following CUDA-specific features are not available in ROCm:
```
Comment (kmaehashi) [inline docs/source/install.rst:432]:
```suggestion
The following features are not available due to the limitation of ROCm or they are specific to CUDA:
```

File: docs/source/install.rst [hunk 3]
```diff
@@ -425,3 +425,41 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
+* Hermitian/symmetric eigenvalue solver (``cupy.linalg.eigh``)
+* Polynomial roots (uses Hermitian/symmetric eigenvalue solver)
+
+The following features may not work in edge cases (e.g., some combinations of dtype):
+
+.. note::
+   We are investigating the root causes of the issues. They are not necessarily
+   CuPy's issues, but ROCm may have some potential bugs.
+
+* 32-bit CArray boundary (HIP is known to fail with sizes 2**32-1024)
```
Comment (kmaehashi) [inline docs/source/install.rst:454]: Maybe move this line to the first section?
I also think it's better to reword this as `Handling extremely large arrays whose size is around 32-bit boundary (HIP is known to fail with sizes 2**32-1024)` as CArray is not familiar word for readers.

File: docs/source/install.rst [hunk 4]
```diff
@@ -425,3 +425,41 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
+* ``cupy.ndarray.__getitem__``
+* ``cupy.cuda.cufft.PlanNd``
+* ``cupy.ix_``
+* Some polynomial routines
+* ``cupy.broadcast``
+* ``cupy.convolve``
+* ``cupy.correlate``
+* Some random sampling routines (``cupy.random``)
+* ``cupy.cublas.dgmm``
```
Comment (kmaehashi) [inline docs/source/install.rst:463]: I think only private APIs can be excluded. (is this called from some API?)

File: docs/source/install.rst [hunk 5]
```diff
@@ -425,3 +425,41 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
+* Polynomial roots (uses Hermitian/symmetric eigenvalue solver)
+
+The following features may not work in edge cases (e.g., some combinations of dtype):
+
+.. note::
+   We are investigating the root causes of the issues. They are not necessarily
+   CuPy's issues, but ROCm may have some potential bugs.
+
+* 32-bit CArray boundary (HIP is known to fail with sizes 2**32-1024)
+* ``cupy.ndarray.__getitem__``
```
Comment (kmaehashi) [inline docs/source/install.rst:455]: Could you add some example failing cases? `__getitem__` sounds too broad.

Comment (takagi) [reply docs/source/install.rst:455]: Sure. I agree that ``__getitem__`` looks too broad. But the failing cases are actually as below and I also don't think  it helps the reader...

https://github.com/cupy/cupy/blob/a652e4707d7a16f431379bf8919c5a47cef4c990/tests/cupy_tests/core_tests/test_ndarray_indexing.py#L81-L94

File: docs/source/install.rst (hunk 4 above)
Comment (takagi) [reply docs/source/install.rst:463]: Let me check it shortly...

File: docs/source/install.rst (hunk 5 above)
Comment (kmaehashi) [reply docs/source/install.rst:455]: As discussed offline, let's add a link to the PR that added this skip code.

File: docs/source/install.rst [hunk 6]
```diff
@@ -425,3 +425,41 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
+* ``cupy.cuda.cufft.PlanNd``
```
Comment (leofang) [inline docs/source/install.rst:456]: I think this is currently private (despite I'd love to make it public)?

File: docs/source/install.rst (hunk 4 above)
Comment (takagi) [reply docs/source/install.rst:463]: `dgmm` is not used from the public APIs so I'll remove the line!

File: docs/source/install.rst (hunk 6 above)
Comment (takagi) [reply docs/source/install.rst:456]: Thanks! I'll remove the line.

File: docs/source/install.rst (hunk 3 above)
Comment (takagi) [reply docs/source/install.rst:454]: Will fix as such!

File: docs/source/install.rst [hunk 7]
```diff
@@ -425,3 +425,39 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
   This behavior is specific to ROCm builds; when building CuPy for NVIDIA CUDA, the build result is not affected by the host configuration.
+
+Limitations
+-----------
+
+The following features are not available due to the limitation of ROCm or because that they are specific to CUDA:
+
+* CUDA Array Interface
+* cuTENSOR
+* Handling extremely large arrays whose size is around 32-bit boundary (HIP is known to fail with sizes 2**32-1024)
```
Comment (kmaehashi) [inline docs/source/install.rst:436]:
```suggestion
* Handling extremely large arrays whose size is around 32-bit boundary (HIP is known to fail with sizes `2**32-1024`)
```

**Comment by kmaehashi** [comment]
LGTM!
pfnCI, test this please.

**Comment by takagi** [comment]
I've also checked the rendered html and it looks fine.

**Comment by kmaehashi** [comment]
Build seems failing?

**Comment by takagi** [comment]
```
/home/runner/work/cupy/cupy/docs/source/install.rst:469:Inline literal start-string without end-string.
```
install.rst doesn't have line 469...

**Comment by takagi** [comment]
I can not reproduce the failure on my local.

File: docs/source/install.rst [hunk 8]
```diff
@@ -425,3 +425,39 @@ You can specify a comma-separated list of ISAs if you have multiple GPUs of diff
...
+
+* ``cupy.ndarray.__getitem__`` (`#4653 <https://github.com/cupy/cupy/pull/4653>`_)
+* ``cupy.ix_`` (`#4654 <https://github.com/cupy/cupy/pull/4654>`_)
+* Some polynomial routines (`#4656 <https://github.com/cupy/cupy/pull/4656>`_, `#4658 <https://github.com/cupy/cupy/pull/4658>`_, `#4758 <https://github.com/cupy/cupy/pull/4758>`_, `#4759 <https://github.com/cupy/cupy/pull/4759>`_)
+* ``cupy.broadcast`` (`#4662 <https://github.com/cupy/cupy/pull/4662>`_)
+* ``cupy.convolve`` (`#4668 <https://github.com/cupy/cupy/pull/4668>`_)
+* ``cupy.correlate`` (`#4781 <https://github.com/cupy/cupy/pull/4781>`_)
+* Some random sampling routines (``cupy.random``, `#4770 <https://github.com/cupy/cupy/pull/4770>`_)
+* ``cupy.linalg.einsum``
+* ``cupyx.scipy.ndimage`` and ``cupyx.scipy.signal``(`#4878 <https://github.com/cupy/cupy/pull/4878>`_, `#4879 <https://github.com/cupy/cupy/pull/4879>`_, `#4880 <https://github.com/cupy/cupy/pull/4880>`_)
```
Comment (emcastillo) [inline docs/source/install.rst:463]: Could it be this?
```suggestion
* ``cupyx.scipy.ndimage`` and ``cupyx.scipy.signal`` (`#4878 <https://github.com/cupy/cupy/pull/4878>`_, `#4879 <https://github.com/cupy/cupy/pull/4879>`_, `#4880 <https://github.com/cupy/cupy/pull/4880>`_)
```

Comment (emcastillo) [reply docs/source/install.rst]: https://techwriter.documatt.com/sphinx-errors/5-inline-literal-start-string-without-end-string.html#solution-to-2

Comment (takagi) [reply docs/source/install.rst]: Thanks!!

**Comment by kmaehashi** [comment]
LGTM!

**Comment by leofang** [comment]
cc: @amathews-amd for vis
